server/service/service_options.go

```
@@ -20,3 +20,60 @@ func (svc service) ModifyOptions(ctx context.Context, req kolide.OptionRequest)
 	}
 	return req.Options, nil
 }
+
+func (svc service) ExpectedCheckinInterval(ctx context.Context) (uint, error) {
+	interval := uint(0)
+	found := false
+
+	osqueryIntervalOptionNames := []string{
+		"distributed_interval",
+		"logger_tls_period",
+	}
+
+	for _, option := range osqueryIntervalOptionNames {
+		// for each option which is known to hold a TLS check-in interval, try to
+		// fetch it
+		opt, err := svc.ds.OptionByName(option)
+		if err != nil {
+			// if the option is not set, try the next known option
+			if _, ok := err.(kolide.NotFoundError); ok {
+				continue
+			}
+			// if some other error occured when getting the option, we want to return
+			// that
+			return 0, err
+		}
+
+		// try to cast the option as a uint. if this fails, the option has likely been set incorrectly
+		var val uint
+		switch v := opt.Value.Val.(type) {
+		case int:
+			val = uint(v)
+		case uint:
+			val = v
+		case uint64:
+			val = uint(v)
+		case float64:
+			val = uint(v)
+		default:
+			return 0, errors.New("Option is not a number: " + opt.Name)
+		}
+
+		// If an option has not been found yet, we want to save this interval.
+		// If an option HAS been found already and this one is less, we want to
+		// save that as our new minimum check-in interval.
+		if !found || val < interval {
+			found = true
+			interval = val
+		}
+	}
+
+	// if we never found any interval options set, the default distributed
+	// interval is 60, so we use that
+	if !found {
+		return 60, nil
```

I wonder if it would be better to return an error here, since this seems to indicate there is a problem with the configuration.

---

Well, this is unlikely to occur because we set the options as apart of migrations, so this should never be hit, but if it was, it would be because those options have not been set and then my logic was that it's safe to default to 60 because that's the osquery default. So, assuming that the option isn't set, default to the default.